packages/unchained-client/src/cosmos/parser/utils.ts

```diff
@@ -108,6 +120,18 @@ const virtualMessageFromEvents = (
       to: msg.to,
       origin: msg.origin
     }
+  } else if (swapEventData) {
+    const sender = swapEventData.attributes.find((attribute) => attribute.key === 'sender')?.value
+    const swapAmount = swapEventData.attributes.find(
+      (attribute) => attribute.key === 'tokens_out'
+    )?.value
+    const valueParsed = swapAmount?.slice(0, swapAmount?.length - 'uosmo'.length) ?? ''
+    if (swapAmount?.includes('uosmo'))
+      return {
+        type: 'send',
```

Comment:
Don't we want to be able to parse Osmo swaps as `TradeType` Txs?
https://github.com/shapeshift/lib/blob/944c8d12dc8380734fe197a8c7555c9c1ce903f7/packages/unchained-client/src/types.ts#L35-L37

Reply:
Im not sure, maybe? The main goal here was to make sure the sends / receives totals of the swap txs are "correct" so the charts are correct

For thorchain we aren't showing them as "trades" either -- they just show up as sends and receives on different chains. It kindof makes sense to me to keep following that same pattern and not worry about displaying any one of the transactions as a "Trade". In my opinion the idea of a "trade" transaction in the parser feels like it was only originally planned for 0x swapper without considering how the others will work

We have talked about the idea of a "blockchain activity" or something more all-encompassing to display trade history. My 2 cents is this is fine until we have a better "trade history" solution for all swappers

Reply:
They could still be marked as trades with applicable metadata, but yes, the concept of a trade occurring across multiple transactions needs some additional thought on how to tie these together and if it can/should be done in the parser

Reply:
Yeah @kaladinlight  as we've discussed with @elmutt I definitely agree this is not a task for now - happy with merging this if it fixes what it's supposed to, unfortunately I didn't get yet the Tx history to show all the Txs it should - not sure yet whether it's a regression from prod or a ws issue

Reply:
@gomesalexandre sounds good and the cosmos stuff is the last main abstraction update that we need to do and I can take some time thinking through this more when we get to that for the parser